docs/BillOfMaterials.md

```diff
@@ -0,0 +1,170 @@
+
+Optional attributes:
+  - `Id` - a unique identifier describing this build.
+  - `Created` - a timestamp of when the build was created.
+
+#### `<Artifacts>`
+
+Describes a group of artifacts
+
+Optional attributes:
+  - `Category` - an arbitrary category name.
+
+#### `<Artifact>`
+
+Describes a file produced by the build.
+
+Required attributes:
+  - `Id` - the file name of the artifact. Id MUST BE unique per each bill of materials.
+  - `Type` - an arbitrary string describing the semantic type of the artifact.
+
+Optional attributes:
+    - Others - any other attributes on the element are considered metadata values about the artifact.
+
+#### `<Dependencies>`
+
+Describes artifact dependencies.
+
+#### `<Link>`
+
+Describes a dependency between two artifacts.
+
+Requried attributes
+  - `Source` - the artifact that has a dependency. This artifact MUST BE defined by an `<Artifact>` in the same `<Build>`
+  - `Target` - the thing it depends on. This artifact MAY OR MAY NOT be described in this `<Build>`.
+    It might have been produced in a previous build or be produced by a partner team.
+
+## Artifact types
+
+These are some common artifact types produced by the build.
+
+  - NuGetPackage
+  - NuGetSymbolsPackage
+  - DebianPackage
+  - RedhatPackage
+  - ZipArchive
+  - TarGzArchive
+  - TextFile
+  - TestBundle
+
+## Artifact categories
+
+Tools may use the category information to define how they handle different files.
+
+These are common categories for artifacts used in aspnet/Universe.
+
+  - ship - used for files that will be published to NuGet.org, microsoft.com, or other official publication locations
+  - shipoob - used for files that will be distributed in other mechanism.
+  - noship - these files should not be published to partner teams or released publically. This may include test artifacts.
+
+## A more complete example
+
+```xml
+<Build Id="(commithash)+Release+Signed+123" Created="12/1/2017 5:10:06 PM +00:00">
+
+    <Artifacts Category="ship">
+        <Artifact Id="Microsoft.Extensions.DependencyInjection.2.0.0-rtm-123.nupkg" Type="NuGetPackage"
+            ShouldBeSigned="true"
+            PackageId="Microsoft.Extensions.DependencyInjection"
+            Version="2.0.0-rtm-123"
+            BranchName="release/2.0.0"
+            CommitHash="xyz123"
+            RepositoryUrl="https://github.com/aspnet/DependencyInjection"
+            FileHash="ABCDEF0123456789" />
+        <Artifact Id="Microsoft.AspNetCore.2.0.0-rtm-123.nupkg" Type="NuGetPackage"
+            ShouldBeSigned="true"
+            PackageId="Microsoft.AspNetCore"
+            Version="2.0.0-rtm-123" />
+        <Artifact Id="Microsoft.AspNetCore.2.0.0-rtm-123.symbols.nupkg" Type="NuGetSymbolsPackage"
+            ShouldBeSigned="true"
+            PackageId="Microsoft.AspNetCore"
+            Version="2.0.0-rtm-123" />
+        <Artifact Id="aspnetcore-store-2.0.0-rtm-123-win-x64.zip" Type="ZipArchive"
```

Comment:
Curious what we need to do to generate these entries.

Reply:
I'm working on an update to BuildTools to generate this. We already generate some of this metadata for nupkgs. https://github.com/aspnet/BuildTools/blob/2fdcbd3f8c1461caed668ebeaf42c9545d30fcbb/files/KoreBuild/modules/solutionbuild/Project.Inspection.targets